plugins/destination/postgresql/client/write.go

```diff
@@ -35,35 +59,38 @@ func (c *Client) Write(ctx context.Context, tables schema.Tables, res <-chan *pl
 		} else {
 			sql = c.upsert(table)
 		}
-
+		batchItems = append(batchItems, &batchItem{
+			table:     r.TableName,
+			sql:       sql,
+			arguments: r.Data,
+		})
 		batch.Queue(sql, r.Data...)
 		if batch.Len() >= c.batchSize {
 			br := c.conn.SendBatch(ctx, batch)
-			if err := br.Close(); err != nil {
-				var pgErr *pgconn.PgError
-				if !errors.As(err, &pgErr) {
-					// not recoverable error
-					return fmt.Errorf("failed to execute batch: %w", err)
+			if batchErr := br.Close(); batchErr != nil {
+				if err := c.findErrorInBatch(ctx, batchItems); err != nil {
+					return err
 				}
-				atomic.AddUint64(&c.metrics.Errors, 1)
-				c.logger.Error().Err(pgErr).Str("table", pgErr.TableName).Msg("failed to execute batch with pgerror")
+				return fmt.Errorf("failed to execute batch and was unable to pinpoint table: %w", batchErr)
```

Comment:
Doesn't this mean everything was successfully inserted? Maybe log it and not return an error